services/src/main/java/org/keycloak/services/resources/account/AccountRestService.java

```diff
@@ -464,6 +450,66 @@ private UserConsentModel createConsent(ClientModel client, ConsentRepresentation
         return consent;
     }
 
+    @Path("/applications")
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    @NoCache
+    public Response applications() {
+        checkAccountApiEnabled();
+        auth.require(AccountRoles.VIEW_APPLICATIONS);
+
+        List<String> activeClients = new LinkedList<String>();
+        List<UserSessionModel> sessions = session.sessions().getUserSessions(realm, user);
+        for (UserSessionModel s : sessions) {
+            activeClients.addAll(s.getAuthenticatedClientSessions().keySet());
+        }
+
+        List<ClientRepresentation> apps = new LinkedList<>();
+        for (ClientModel client : realm.getClients()) {
```

Comment:
I'm concerned with performance of realm.getClients if there are thousands of clients. We have an open PR for pagination of clients in admin console for this reason.

Ideal would be to add a query to obtain clients that should be advertised to a users in account console. The applications should be clients that:

* Are not bearer-only
* Have baseUrl set
* Have the option "always show in account console" enabled. This was discussed on the ML as the simplest way to have an option for admins to decide if a client should be advertised or not in the account console. Alternative of displaying only clients where a user has at least one client role would be too expensive and complex, so we should just do a simple option instead to explicitly advertise clients not current in-use

Reply:
This needs to be changed as it won't scale and also will display many clients we don't want to display.

The logic should probably be something like:

* Fetch sessions - build a list of clients from sessions
* Fetch persisted user consents - build a list of clients from persisted consents combining with above
* Fetch clients with "always show in account console" - combine with above